{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "dbc6a1ec-7b74-4a8e-9d5b-dbb148338f9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "from openai import OpenAI\n",
    "from pydantic import BaseModel, Field, conlist\n",
    "import time\n",
    "import numpy as np\n",
    "from numpy.linalg import norm\n",
    "from scipy.interpolate import UnivariateSpline\n",
    "from matplotlib import pyplot as plt\n",
    "import re\n",
    "from IPython.display import clear_output\n",
    "import sys\n",
    "from rake_nltk import Rake\n",
    "import nltk\n",
    "\n",
    "client = OpenAI(api_key = os.environ[\"OPENAI_API_KEY\"])\n",
    "\n",
    "class gen_format(BaseModel):\n",
    "    Question: conlist(str, min_length=5, max_length=5) = Field(description=\"A list of five questions that could be answered by the given answer\")\n",
    "    \n",
    "class rag_format(BaseModel):\n",
    "    Ranked_Relevant_Information: list[str] = Field(description=\"The ranked list of information relevant to the query, with no paraphrasing or changing of the text from the files.\")\n",
    "\n",
    "def preprocess_text(text):\n",
    "    # Replace decimals/commas in numbers with an underscore and replace hyphens with underscores, generally (except for negative numbers).\n",
    "    #It is only these cases that the sentence tokenizer in Rake doesn't seem to handle well\n",
    "    text = re.sub(r'(\\d+)\\.(\\d+)', r'\\1_\\2', text)\n",
    "    text = re.sub(r'(\\d+)\\,(\\d+)', r'\\1\\2', text)\n",
    "    # Pattern explanation:\n",
    "    # (?<!\\s)-(?!\\d) - matches hyphens not preceded by whitespace or followed by digit\n",
    "    # | - OR\n",
    "    # (?<=\\s)-(?=\\D) - matches hyphens preceded by whitespace and followed by non-digit\n",
    "    text = re.sub(r'(?<!\\s)-(?!\\d)|(?<=\\s)-(?=\\D)', '_', text)\n",
    "    return text\n",
    "    \n",
    "def embedding_answers(answer, ideal, custom_stopwords, english_words) -> str:\n",
    "    #tell Rake to leave logical comparatives alone\n",
    "    r = Rake(stopwords=custom_stopwords)\n",
    "    # Extraction given the text.\n",
    "    text1=preprocess_text(answer)\n",
    "    text2=preprocess_text(ideal)\n",
    "    r.extract_keywords_from_text(text1)\n",
    "    key_phrases1=r.get_ranked_phrases()\n",
    "    r.extract_keywords_from_text(text2)\n",
    "    key_phrases2=r.get_ranked_phrases()\n",
    "    print(key_phrases1)\n",
    "    print(key_phrases2)\n",
    "    result_1=[]\n",
    "    i=0\n",
    "    indicies=[]\n",
    "    for string_ideal in key_phrases2:\n",
    "        #check for \"names\" that need to be matched exactly\n",
    "        #checks that string_ideal is one word with at least one letter and that is not in english\n",
    "        if (not (\" \" in string_ideal)) and (any(char.isalpha() for char in string_ideal)) and (not (string_ideal in english_words)):\n",
    "            #if this word does exist in the answer...\n",
    "            if (string_ideal in text1.lower()):\n",
    "                #we have a match!\n",
    "                result_1.append(1)\n",
    "            else:\n",
    "                #if not, no match, therefore \"incorrect\"\n",
    "                result_1.append(0.7)\n",
    "        else:\n",
    "            max_cos=0\n",
    "            resp1 = client.embeddings.create(\n",
    "                input=string_ideal,\n",
    "                model=\"text-embedding-ada-002\",\n",
    "                encoding_format= \"float\",\n",
    "            )\n",
    "            check=0\n",
    "            j=0\n",
    "            indicies.append(0)\n",
    "            for string_gen in key_phrases1:\n",
    "                if (string_ideal==string_gen and check==0):\n",
    "                    max_cos=1\n",
    "                    result_1.append(max_cos)\n",
    "                    indicies[i]=j\n",
    "                    check=1\n",
    "                j+=1\n",
    "            if (max_cos!=1):\n",
    "                j=0\n",
    "                for string_gen in key_phrases1:\n",
    "                    resp2 = client.embeddings.create(\n",
    "                        input=string_gen,\n",
    "                        model=\"text-embedding-ada-002\",\n",
    "                        encoding_format= \"float\",\n",
    "                    )\n",
    "                    a=np.array(resp1.data[0].embedding)\n",
    "                    b=np.array(resp2.data[0].embedding)\n",
    "                    cos=np.dot(a,b)/(norm(a)*norm(b))\n",
    "                    if (cos>max_cos):\n",
    "                        max_cos=cos\n",
    "                        indicies[i]=j\n",
    "                    j+=1\n",
    "                result_1.append(max_cos)\n",
    "            i+=1\n",
    "    return np.array(result_1)\n",
    "\n",
    "def embedding_questions(question, answer, question_model) -> str:\n",
    "    og_question = client.embeddings.create(\n",
    "        input=question,\n",
    "        model=\"text-embedding-ada-002\",\n",
    "        encoding_format= \"float\",\n",
    "    )\n",
    "    gen_message=\"\"\"\n",
    "    You are a question generating agent. Your task is to generate a list of questions for the given answer.\n",
    "    \"\"\"\n",
    "    gen_assistant = client.beta.assistants.create(\n",
    "        name=\"gen_test\",\n",
    "        instructions=gen_message,\n",
    "        model=question_model, \n",
    "        temperature = 0.0,\n",
    "        top_p = 0.2,\n",
    "        response_format= {\n",
    "            \"type\": \"json_schema\",\n",
    "            \"json_schema\": {\n",
    "                \"name\": \"answer\",\n",
    "                \"schema\": gen_format.model_json_schema()\n",
    "            },\n",
    "        }\n",
    "    )\n",
    "    \n",
    "    thread = client.beta.threads.create(\n",
    "                    messages=[],\n",
    "                )\n",
    "    \n",
    "    parsed = client.beta.threads.messages.create(\n",
    "                    thread_id=thread.id,\n",
    "                    content=answer,\n",
    "                    role='user',\n",
    "                )\n",
    "    \n",
    "    run = client.beta.threads.runs.create(\n",
    "        thread_id=thread.id,\n",
    "        assistant_id=gen_assistant.id,\n",
    "        # pass the latest system message as instructions\n",
    "        instructions=gen_message,\n",
    "    )\n",
    "    run = client.beta.threads.runs.retrieve(run.id, thread_id=thread.id)\n",
    "    while run.status!=\"completed\":\n",
    "        run = client.beta.threads.runs.retrieve(run.id, thread_id=thread.id)\n",
    "    response_messages = client.beta.threads.messages.list(thread.id, order=\"asc\")\n",
    "    for message in response_messages.data:\n",
    "        for content in message.content:\n",
    "            output=content.text.value\n",
    "            if output.startswith(\"{\"):\n",
    "                data=json.loads(output)\n",
    "                generated=data[\"Question\"]\n",
    "    client.beta.assistants.delete(assistant_id=gen_assistant.id)\n",
    "    result=[]\n",
    "    for gen_question in generated:\n",
    "        gen_question_vector = client.embeddings.create(\n",
    "            input=gen_question,\n",
    "            model=\"text-embedding-ada-002\",\n",
    "            encoding_format= \"float\",\n",
    "        )\n",
    "        a=np.array(gen_question_vector.data[0].embedding)\n",
    "        b=np.array(og_question.data[0].embedding)\n",
    "        result.append(np.dot(a, b)/(norm(a)*norm(b)))\n",
    "    return result\n",
    "    \n",
    "def embedding_search(vector_store, question, answer, search_model) -> str:\n",
    "    rag_message=\"\"\"You are a retrieval agent tasked with performing file searches to find information for the purpose of providing answers.\n",
    "        Find pieces of information that will be directly relevant for answering the query and rank these pieces of information from most relevant to least relevant.\n",
    "        You must quote the passages from the files directly. Do not paraphrase or change the text in any way.\n",
    "        Do not add anything else to the passage quotations, including sources and filenames.\n",
    "        If no information is relevant, you must return a single piece of information, where you state \"No information found\".\n",
    "        Ideally, these pieces of information will be sentences, phrases, data points or sets of data points, but you have limited flexiblility to include other pieces of information if you think they are appropriate.\n",
    "        \n",
    "        You must use tool call (i.e., file search).\n",
    "        \n",
    "        You know about the content of the code-base.\n",
    "        \"\"\"\n",
    "    rag_assistant = client.beta.assistants.create(\n",
    "        name=\"rag_test\",\n",
    "        instructions=rag_message,\n",
    "        tools=[\n",
    "            {\"type\": \"file_search\",\n",
    "                \"file_search\":{\n",
    "                    'max_num_results': 10,\n",
    "                    \"ranking_options\": {\n",
    "                        \"ranker\": \"auto\",\n",
    "                        \"score_threshold\": 0.6\n",
    "                    }\n",
    "                }\n",
    "            }\n",
    "        ],\n",
    "        tool_resources={\"file_search\": {\"vector_store_ids\":[vector_store.id]}},\n",
    "        model=search_model, \n",
    "        temperature = 0,\n",
    "        top_p = 0.2,\n",
    "        response_format= {\n",
    "            \"type\": \"json_schema\",\n",
    "            \"json_schema\": {\n",
    "                \"name\": \"answer\",\n",
    "                \"schema\": rag_format.model_json_schema()\n",
    "            },\n",
    "        }\n",
    "    )\n",
    "    thread = client.beta.threads.create(\n",
    "                    messages=[],\n",
    "                )\n",
    "    \n",
    "    parsed = client.beta.threads.messages.create(\n",
    "                    thread_id=thread.id,\n",
    "                    content=question,\n",
    "                    role='user',\n",
    "                )\n",
    "    \n",
    "    run = client.beta.threads.runs.create(\n",
    "        thread_id=thread.id,\n",
    "        assistant_id=rag_assistant.id,\n",
    "        # pass the latest system message as instructions\n",
    "        instructions=rag_message,\n",
    "    )\n",
    "    run = client.beta.threads.runs.retrieve(run.id, thread_id=thread.id)\n",
    "    while run.status!=\"completed\":\n",
    "        run = client.beta.threads.runs.retrieve(run.id, thread_id=thread.id)\n",
    "    response_messages = client.beta.threads.messages.list(thread.id, order=\"asc\")\n",
    "    for message in response_messages.data:\n",
    "        for content in message.content:\n",
    "            output=content.text.value\n",
    "            if output.startswith(\"{\"):\n",
    "                data=json.loads(output)\n",
    "                try:\n",
    "                    information=data[\"Ranked_Relevant_Information\"]\n",
    "                except:\n",
    "                    information=data[\"Ranked Relevant Information\"]\n",
    "    if (\"information\" in locals()):\n",
    "        #uncomment for hallucination guarding\n",
    "        \"\"\"run_steps = client.beta.threads.runs.steps.list(\n",
    "            thread_id=thread.id,\n",
    "            run_id=run.id\n",
    "        )\n",
    "        j=0\n",
    "        for step in run_steps.data:\n",
    "            #wait until the runs.steps.list has finished\n",
    "            while step.status!=\"completed\":\n",
    "                run_steps = client.beta.threads.runs.steps.list(\n",
    "                    thread_id=thread.id,\n",
    "                    run_id=run.id\n",
    "                )\n",
    "            if (j!=0):\n",
    "                retrieved_step = client.beta.threads.runs.steps.retrieve(\n",
    "                    thread_id=step.thread_id,\n",
    "                    run_id=run.id,\n",
    "                    step_id=step.id,\n",
    "                    include=[\"step_details.tool_calls[*].file_search.results[*].content\"]\n",
    "                )\n",
    "                #check for hallucinations and flag all \"offending text\" from the passed information list\n",
    "                information=hallucination_check(retrieved_step, information, question, 20)\n",
    "            j+=1\n",
    "        \"\"\"\n",
    "        pass\n",
    "    else:\n",
    "        information=[\"No information.\"]\n",
    "    answer_resp = client.embeddings.create(\n",
    "        input=answer,\n",
    "        model=\"text-embedding-ada-002\",\n",
    "        encoding_format= \"float\",\n",
    "    )\n",
    "    results=[]\n",
    "    for i in range(len(information)):\n",
    "        info_resp = client.embeddings.create(\n",
    "            input=information[i],\n",
    "            model=\"text-embedding-ada-002\",\n",
    "            encoding_format= \"float\",\n",
    "        )\n",
    "        a=np.array(info_resp.data[0].embedding)\n",
    "        b=np.array(answer_resp.data[0].embedding)\n",
    "        results.append(np.dot(a, b)/(norm(a)*norm(b)))\n",
    "    i=0\n",
    "    #add is binary True/False\n",
    "    add=1\n",
    "    my_diff=0\n",
    "    mean_score=[]\n",
    "    for score in results:\n",
    "        if (i!=0):\n",
    "            my_diff=np.absolute(score-store)\n",
    "            if (i==1):\n",
    "                ref_diff=my_diff\n",
    "            if (my_diff>=2*ref_diff):\n",
    "                add=0\n",
    "        if (add==1):\n",
    "            mean_score.append(score)\n",
    "            store=score\n",
    "            i+=1\n",
    "    return mean_score\n",
    "def hallucination_check(retrieved_step, information, question, char_match):\n",
    "    raw_information=\"\"\n",
    "    for result in retrieved_step.step_details.tool_calls[0].file_search.results:\n",
    "        raw_information=raw_information+result.content[0].text\n",
    "    #regex expression to remove all references from the text before \"cleaning\".\n",
    "    raw_information = re.sub(r\"\\(\\d+\\)|\\[\\d+\\]|\\(.*Fig.*\\)|\\(.*Table.*\\)\", \"\", raw_information)\n",
    "    file_information=''.join(ch for ch in raw_information if ch.isalnum())\n",
    "    for i in range(len(information)):\n",
    "        verify=information[i].split(\"...\")\n",
    "        for split in verify:\n",
    "            filtered_information=''.join(ch for ch in split if ch.isalnum())\n",
    "            if not (filtered_information in file_information):\n",
    "                clear_output(wait=True)\n",
    "                print(\"AI is answering this question:\\n\"+question)\n",
    "                print(\"AI understood:\\n\"+split)\n",
    "                length=len(filtered_information)\n",
    "                if length < char_match:\n",
    "                    char_match = length\n",
    "                index=file_information.find(filtered_information[:char_match])\n",
    "                if (index!=-1):\n",
    "                    print(\"The file contained:\\n\"+file_information[index:index+length])\n",
    "                else:\n",
    "                    index=file_information.find(filtered_information[length-char_match:])\n",
    "                    if (index!=-1):\n",
    "                        print(\"The file contained:\\n\"+file_information[index-length+char_match:index+char_match])\n",
    "                    else:\n",
    "                        print(\"The file contained:\\n\"+raw_information)\n",
    "                print(\"Waiting for user input...\")\n",
    "                print(\"Correct the potential hallucination. If the AI is correct, type 'y'. If information given by the AI is not in the file segment, type 'n'.\", end='', flush=True)\n",
    "                user=input()\n",
    "                print(\"User entered:\", user)\n",
    "                if (user!='y'):\n",
    "                    print(\"The file contained:\\n\"+raw_information)\n",
    "                    print(\"Correct the potential hallucination. If the AI is correct, type 'y'. If no information from these chunks is relevant, type 'n'. Otherwise, input what the information should be.\", end='', flush=True)\n",
    "                    user=input()\n",
    "                    if (user!='y'):\n",
    "                        if(user=='n'):\n",
    "                            split=\"Null information.\"\n",
    "                        else:\n",
    "                            split=user\n",
    "        information[i]=\"\"\n",
    "        for split in verify:\n",
    "            information[i]=information[i]+\"...\"+split\n",
    "        information[i]=information[i][3:]\n",
    "    return information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "6bd0d5c4-f466-4db4-a3fc-37c9680c499f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to /home/adrian/nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "[nltk_data] Downloading package words to /home/adrian/nltk_data...\n",
      "[nltk_data]   Package words is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "nltk.download('stopwords')\n",
    "nltk.download('words')\n",
    "english_words = set(nltk.corpus.words.words())\n",
    "\n",
    "custom_stopwords = set(nltk.corpus.stopwords.words('english')) - {\"no\", \"not\", \"than\", \"more\", \"same\", \"before\", \"after\", \"now\", \"then\", \"above\", \"below\", \"over\", \"under\", \"like\", \"other\", \"such\", \"few\", \"most\", \"some\", \"between\"}  # Keep logical comparatives- important for RAG analysis \n",
    "\n",
    "\n",
    "assistant_data = \"/home/adrian/Documents/University Work/Part III Project/PaperQA2/LitQA2_Papers\"\n",
    "lit = pd.read_csv('../PaperQA2/LitQA2_edit.csv')\n",
    "question=[]\n",
    "for i in range(lit.shape[0]):\n",
    "    question.append(lit.loc[i, \"question\"])\n",
    "#extract answer from these agent and ideal from LitQA2\n",
    "with open(\"output_4o_mini_exact_extended.txt\", 'r', encoding='utf-8') as file:\n",
    "    file_content = file.read()\n",
    "spaces=0\n",
    "tab=0\n",
    "i=0\n",
    "answer=[]\n",
    "ideal=[]\n",
    "for char in file_content:\n",
    "    if (spaces==2 and char != \"\\t\" and tab==0):\n",
    "        answer[i]+=char\n",
    "    elif (char ==\"\\t\"):\n",
    "        spaces=0\n",
    "        ideal.append(\"\")\n",
    "        tab+=1\n",
    "    elif (char != \"\\n\" and tab==1):\n",
    "        ideal[i]+=char\n",
    "    elif (char == \"\\n\" and tab==1):\n",
    "        i+=1\n",
    "        tab=0\n",
    "    if (char == \" \" and spaces<2 and tab==0):\n",
    "        spaces+=1\n",
    "        if (spaces==2):\n",
    "            answer.append(\"\")\n",
    "for i in range(len(answer)):\n",
    "    answer[i]=\" \".join(word for word in answer[i].split() if \".pdf\" not in word)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "ff78eba2-3ee0-4e38-83c9-80dbb2403bbd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Embedding Answers\n",
      "['acinetobacter lwoffii', 'resistant', 'lab', 'evolved', 'ciprofloxacin']\n",
      "['ciproflaxin']\n",
      "Acinetobacter lwoffii has evolved in the lab to be resistant to ciprofloxacin.\tciproflaxin [0.7]\n",
      "\n",
      "['active olfactory receptor genes increase', 'mouse olfactory neurons', 'greek island regions', 'factor', 'contacts', '2_7_fold']\n",
      "['2_7 fold']\n",
      "Active olfactory receptor genes increase their contacts with Greek island regions by a factor of 2.7-fold in mouse olfactory neurons.\t2.7 fold [0.97913536]\n",
      "\n",
      "['approximately 12309 unique transposable element insertion loci', 'citrus reticulata', 'mandarin', 'genome']\n",
      "['12300']\n",
      "There are approximately 12,309 unique transposable element insertion loci in the mandarin (Citrus reticulata) genome.\t12300 [0.81095288]\n",
      "\n",
      "['h3_3k36r mutation finish developing', 'approximately 80', 'eclose', 'drosophila']\n",
      "['80']\n",
      "Approximately 80% of Drosophila with a H3.3K36R mutation finish developing and eclose.\t80% [0.88708404]\n",
      "\n",
      "['many rna polymerase ii promoter sequences', 'suggest enhancers influence gene expression', 'enhance rna output', 'enhancers varies significantly', 'enhancer_promoter class compatibilities', 'promoter sequences', 'promoter activity', 'enhancer sequences', 'smaller factor', 'model differs', 'human cells', 'distinct sets', 'different types', 'broadly compatible', '3d interactions', 'enhancer activity', 'enhancers', 'enhancer', 'work', 'supported', 'responsiveness', 'promoters', 'observations', 'involved', 'however', 'genes', 'drosophila', 'compatibility', 'combination', 'cofactors']\n",
      "['yes', 'no']\n",
      "Enhancer and promoter sequences are broadly compatible in human cells, as enhancers can work with many RNA polymerase II promoter sequences to enhance RNA output. This compatibility is supported by observations that suggest enhancers influence gene expression through a combination of promoter activity, enhancer activity, and 3D interactions, with enhancer-promoter class compatibilities being a smaller factor. However, this model differs from Drosophila, where the responsiveness of promoters to enhancers varies significantly, with distinct sets of enhancer sequences and cofactors being involved for different types of genes.\tyes, no [0.81908473 0.8233194 ]\n",
      "\n",
      "['tumor_infiltrating leukocytes', 'zt13', 'tils', 'specifically', 'number', 'highest', 'evening']\n",
      "['evening']\n",
      "The number of tumor-infiltrating leukocytes (TILs) is highest in the evening, specifically at ZT13.\tevening [1]\n",
      "\n",
      "['single_nuclei rna sequencing data', 'cell – matrix interactions', 'specifically including col22a1', 'myotendinous junction myonuclei', 'genes involved', 'enriched', 'based', 'adgbr4']\n",
      "['cell – matrix interactions']\n",
      "Based on single-nuclei RNA sequencing data, myotendinous junction myonuclei are enriched for genes involved in cell–matrix interactions, specifically including COL22A1 and ADGBR4.\tcell–matrix interactions [1]\n",
      "\n",
      "['no information']\n",
      "['5_3']\n",
      "No information.\t5.3% [0.77573453]\n",
      "\n",
      "['highest yield', 'beta_amyrin synthase', 'beta_amyrin', 'vaccaria', 'svbas', 'squalene', 'produces']\n",
      "['saponaria vaccaria']\n",
      "The beta-amyrin synthase from S. vaccaria (SvBAS) produces the highest yield of beta-amyrin from squalene.\tSaponaria vaccaria [0.87005086]\n",
      "\n",
      "['immobilize dendritic cell membranes', 'dendritic cell vaccines', 'cell priming capabilities', 'metabolic glycan labeling', 'dcs ), activates', 'dendritic cells', 'membrane mobility', 'antitumor efficacy', 'antigen presentation', 'yes', 'reduces', 'improves', 'enhances']\n",
      "['yes']\n",
      "Yes, metabolic glycan labeling does immobilize dendritic cell membranes and enhances the antitumor efficacy of dendritic cell vaccines. It reduces the membrane mobility of dendritic cells (DCs), activates them, and improves their antigen presentation and T cell priming capabilities.\tYes [1]\n",
      "\n",
      "['more than twofold higher than', 'average across multiple crispri screens', 'sgrnas targeting coding strands', 'higher effect sizes', 'average effect sizes', 'gene bodies leads', 'targeting template strands', 'coding strands', 'targeting sgrnas', 'sgrnas targeting', 'coding strand', 'template strand', 'significant impact', 'targeting', 'yes', 'specifically', 'not', 'growth', 'additionally']\n",
      "['coding strand']\n",
      "Yes, targeting sgRNAs to coding strands in gene bodies leads to higher effect sizes on average across multiple CRISPRi screens. Specifically, the average effect sizes of sgRNAs targeting coding strands were more than twofold higher than those targeting template strands. Additionally, sgRNAs targeting the coding strand had a significant impact on growth, while those targeting the template strand did not.\tcoding strand [1]\n",
      "\n",
      "['disruptions included enlarged granular cells', 'also caused significant cellular disruptions', 'n_terminal yeast αprepro sequence', 'αprepro sequence increased sensitivity', 'yeast cell phenotype', 'yeast cell phenotype', 'human mu_opioid receptor', 'expanded endoplasmic reticulum', 'agonists like damgo', '72 %, indicating', 'overall signaling competency', 'αprepro sequence', 'cellular stress', 'signaling population', 'improved sensitivity', 'ultimately led', 'roughly two_fold', 'potential misfolding', 'negative effects', 'maximum size', 'dramatically decreased', 'compromised due', 'complex effect', 'sensitivity', 'therefore', 'relocalization', 'puncta', 'morphine', 'membranes', 'hsmor_gfp', 'hsmor', 'hsmor', 'expression', 'er', 'despite', 'beneficial', 'additionally']\n",
      "['enlarged granular cells']\n",
      "The expression of human mu-opioid receptor (HsMOR) with an N-terminal yeast αPrePro sequence has a complex effect on yeast cell phenotype. While the αPrePro sequence increased sensitivity to agonists like DAMGO and morphine by roughly two-fold, it also caused significant cellular disruptions. These disruptions included enlarged granular cells, expanded endoplasmic reticulum (ER) membranes, and the relocalization of HsMOR-GFP to puncta. Additionally, the maximum size of the signaling population was dramatically decreased by 72%, indicating that despite the improved sensitivity, the overall signaling competency was compromised due to cellular stress and potential misfolding of HsMOR. Therefore, the αPrePro sequence, while beneficial for sensitivity, ultimately led to negative effects on the yeast cell phenotype.\tEnlarged granular cells [0.94484935]\n",
      "\n",
      "['homology based structure predicted', 'extracellular tm2_tm3 loop region', 'not accurately predict', 'extracellular end', 'poor prediction', 'partial unwinding', 'tm3', 'colabfold']\n",
      "['tm3']\n",
      "The homology based structure predicted by ColabFold has a poor prediction for the extracellular TM2-TM3 loop region and does not accurately predict the partial unwinding of TM3 at the extracellular end.\tTM3 [1]\n",
      "\n",
      "['mycobacterial repressor darr dimer results', 'approximately 1300 å²', 'surface', 'solvent', 'formation', 'burial']\n",
      "['1300 anstrom', '2']\n",
      "The formation of the mycobacterial repressor DarR dimer results in the burial of approximately 1300 Å² of surface from the solvent.\t1300 anstrom^2 [0.85564181 0.81999364]\n",
      "\n",
      "['adenosine a3 receptor', 'a3ar antagonist cf101', 'ecl3 region', 'a2aar increases', 'a3ar', 'onto', 'grafting', 'efficacy', 'binding']\n",
      "['increases']\n",
      "Grafting the ECL3 region from the adenosine A3 receptor (A3AR) onto the A2AAR increases the efficacy of binding to the A3AR antagonist CF101.\tincreases [1]\n",
      "\n",
      "['method revealed similar gene detection', 'several time_consuming steps such', 'pcr mix without cleaning', 'demonstrated robust tagmentation over', 'certain pcr polymerases', 'showed improved compatibility', 'biased towards 5', 'better library quality', 'excessive cdna preamplification', 'direct tagmentation', 'cdna first', 'cdna directly', 'cdna amounts', 'resulting libraries', 'reaction volumes', 'particularly seqamp', 'full_length coverage', 'concentration measurements', 'tagmentation', 'yes', 'transcripts', 'smart_seq3xpress', 'referred', 'reads', 'range', 'quantifying', 'noted', 'leading', 'however', 'found', 'exclusion', 'despite', 'complexity', 'challenges', 'assessment', 'approach', 'allowing', 'affected']\n",
      "['yes']\n",
      "Yes, there has been an assessment of doing tagmentation of cDNA directly in the PCR mix without cleaning and quantifying the cDNA first. This approach, referred to as Smart-seq3xpress, demonstrated robust tagmentation over a range of cDNA amounts and reaction volumes, allowing for the exclusion of several time-consuming steps such as excessive cDNA preamplification and concentration measurements. However, it was noted that while this method revealed similar gene detection, the resulting libraries were biased towards 5' reads, which affected the full-length coverage of transcripts. Despite these challenges, it was found that certain PCR polymerases, particularly SeqAmp, showed improved compatibility with direct tagmentation, leading to better library quality and complexity.\tYes [1]\n",
      "\n",
      "['layer 6 neurons exhibit', 'include lower layer 5', 'layer 6 intratelencephalic neurons', 'layer 6 neurons', 'lower layer 5', 'lower layer 5', '4 intratelencephalic neurons', 'project more diffusely', 'more diffuse compared', '4 neurons', 'layer 2', 'layer 2', 'upper somata', 'smaller area', 'laminar patterns', 'greater degree', 'deep somata', 'axonal terminations', 'axonal terminations', 'axonal terminations', 'therefore', 'tend', 'projections', 'observation', 'indicated', 'focus', 'diffusion', 'correspond']\n",
      "['more diffuse']\n",
      "The laminar patterns of axonal terminations of lower Layer 5/Layer 6 intratelencephalic neurons are more diffuse compared to Layer 2-4 intratelencephalic neurons. This is indicated by the observation that deep somata (which include lower Layer 5/Layer 6 neurons) tend to project more diffusely, while upper somata (which correspond to Layer 2-4 neurons) focus their axonal terminations in a smaller area. Therefore, the axonal terminations of lower Layer 5/Layer 6 neurons exhibit a greater degree of diffusion in their projections.\tMore diffuse [0.945713]\n",
      "\n",
      "['negative bias persisted even after', 'subgenual anterior cingulate cortex', 'decision_making metrics remained above', 'negative effective sites compared', 'negative effective sessions', 'mean percentage change', 'subsequent decision_making tasks', 'influence decision_making negatively', 'follow_up decision_making experiments', 'microstimulation successfully led', 'negative bias', 'effective microstimulation', 'significant results', 'positive ones', 'greater proportion', 'decision_making', 'microstimulation', 'threshold', 'sustained', 'suggesting', 'specifically', 'shown', 'sgacc', 'sgacc', 'monkeys', 'microstimulations', 'induce', 'indicating', 'furthermore', 'evidenced', 'ended', 'effects', 'activity']\n",
      "['negative bias', 'maintain']\n",
      "Microstimulations in the subgenual anterior cingulate cortex (sgACC) of monkeys have been shown to induce a negative bias in decision-making. Specifically, the microstimulation successfully led to a greater proportion of negative effective sites compared to positive ones, indicating that the sgACC's activity can influence decision-making negatively. Furthermore, this negative bias persisted even after the effective microstimulation had ended, as evidenced by significant results in follow-up decision-making experiments. The mean percentage change in decision-making metrics remained above the threshold for negative effective sessions, suggesting that the effects of the microstimulation were sustained in subsequent decision-making tasks.\tThey maintain a negative bias [1.         0.86408751]\n",
      "\n",
      "['1 more strongly than', 'human snvs french 2', 'ets_a site increases', 'transcription factor ets', 'french 2', 'wild_type ets_a', 'indian 2', 'indian 2', '1 due', 'preaxial polydactyly', 'mechanism driving', 'enhanced binding', 'emsa confirms', 'condition may', 'variants bind', 'subtle increase', 'relative affinity', '1', 'ets', 'ets', 'variants', 'increase', 'affinity', 'affinity', 'suggesting', 'specifically', 'reference', 'phenotype', 'linked', 'dna', 'cause', 'associated', 'additionally', '0_26', '0_24', '0_15']\n",
      "['subtle increase']\n",
      "The human SNVs French 2 and Indian 2 both cause a subtle increase in the relative affinity of the transcription factor ETS-1 to DNA. Specifically, the affinity for the ETS-A site increases from 0.15 in the reference to 0.24 in French 2 and 0.26 in Indian 2. Additionally, EMSA confirms that both variants bind to ETS-1 more strongly than the wild-type ETS-A. This increase in affinity is associated with the phenotype of preaxial polydactyly, suggesting that the mechanism driving this condition may be linked to the enhanced binding of ETS-1 due to these variants.\tsubtle increase [1]\n",
      "\n",
      "['dnmt – tko neurons reflects changes seen', 'dnmt1 −/− embryonic stem cells', 'then comprise one_third', 'similar derepression effects', 'total repeat_derived rna', 'repeat_derived rna expression', 'neurons leads', 'repeat_derived rna', 'repeat_derived rna', 'other contexts', 'neighboring genes', 'murine embryos', 'dna methyltransferase', 'also impact', '200_fold induction', 'dramatic increase', 'expression', 'increase', 'vivo', 'upregulation', 'such', 'results', 'particularly', 'particle', 'observed', 'knocking', 'intracisternal', 'indicating', 'iaps', 'iap', 'elements']\n",
      "['strong increase']\n",
      "Knocking out DNA methyltransferase in neurons leads to a dramatic increase in repeat-derived RNA expression, particularly from IAP (intracisternal A particle) elements. This results in a 200-fold induction of IAPs, which then comprise one-third of the total repeat-derived RNA. This increase in repeat-derived RNA can also impact the expression of neighboring genes. Similar derepression effects have been observed in other contexts, such as in Dnmt1−/− embryonic stem cells and murine embryos, indicating that the upregulation of repeat-derived RNA in DNMT–TKO neurons reflects changes seen in vivo.\tstrong increase [0.91272967]\n",
      "\n",
      "['conformational changes induced', 'wip1 phosphatase significantly', 'threonine dephosphorylation', 'substantial stimulation', 'pt180 dephosphorylation', 'position t180', 'phosphorylated threonine', 'more accessible', 'activity due', 'activation loop', 'pexmetinib increases', 'pexmetinib enhances', 'wip1', 'wip1', 'pexmetinib', 'specifically', 'solvent', 'rate', 'rate', 'p38α', 'making', 'indicating', 'flipping', 'exposes', 'dephosphorylate', '2p', '10_fold']\n",
      "['increases dephosphorylation']\n",
      "Pexmetinib enhances the rate of threonine dephosphorylation by WIP1 phosphatase significantly. It does this by flipping the activation loop of p38α-2p, which exposes the phosphorylated threonine at position T180 to solvent, making it more accessible for WIP1 to dephosphorylate. Specifically, pexmetinib increases the rate of pT180 dephosphorylation by 10-fold, indicating a substantial stimulation of WIP1's activity due to the conformational changes induced by pexmetinib.\tIncreases dephosphorylation [0.92442025]\n",
      "\n",
      "['snipr001 phage therapy comprised 72 fluoroquinolone_resistant e', 'panel used', 'coli strains', 'evaluate']\n",
      "['72']\n",
      "The panel used to evaluate the SNIPR001 phage therapy comprised 72 fluoroquinolone-resistant E. coli strains.\t72 [0.79259462]\n",
      "\n",
      "['marked cre_dependent neuronal death observed', '2 weeks following crispr inactivation', 'primary neurons cultured', 'mouse neurons survive', 'lsl_crispri mice', 'indicated', 'hspa5']\n",
      "['14 days']\n",
      "Mouse neurons survive for about 2 weeks following CRISPR inactivation of HSPA5, as indicated by marked Cre-dependent neuronal death observed in primary neurons cultured from LSL-CRISPRi mice.\t14 days [0.770611]\n",
      "\n",
      "['two fmd cycles', 'significant delay', 'minimum required', 'melanoma tumors', 'mice', 'growth', 'cause']\n",
      "['2']\n",
      "The minimum required to cause a significant delay in the growth of melanoma tumors in mice is two FMD cycles.\t2 [0.80238927]\n",
      "\n",
      "['20 unique structural clades', 'structure_based clustering', 'adenosine deaminases', 'according']\n",
      "['1']\n",
      "According to structure-based clustering, there are 20 unique structural clades of adenosine deaminases.\t1 [0.80773907]\n",
      "\n",
      "['18164 distinct sites containing tng_repeat_like elements', 'homo sapiens ),', 'human genome']\n",
      "['18164']\n",
      "In the human genome (Homo sapiens), there are 18,164 distinct sites containing TnG-repeat-like elements.\t18164 [0.80995624]\n",
      "\n",
      "['5765 genes show changes', 'promoter regions compared', 'braak stages v', '5mc methylation', 'vi', 'patients', 'control', 'alzheimer']\n",
      "['5765']\n",
      "In Alzheimer's patients at Braak stages V/VI, 5,765 genes show changes in 5mC methylation of their promoter regions compared to control.\t5765 [0.8612301]\n",
      "\n",
      "['murine brown adipocytes', '7875 phosphorylation sites', 'treated', 'regulated', 'forskolin']\n",
      "['7875']\n",
      "When treated with forskolin, 7,875 phosphorylation sites were regulated in murine brown adipocytes.\t7875 [0.87849574]\n",
      "\n",
      "['5_fold more sensitive', 'kate mutant strain', 'wild_type strain', 'desiccation than', 'acinetobacter baumanii']\n",
      "['5_fold']\n",
      "A katE mutant strain of Acinetobacter baumanii is about 5-fold more sensitive to desiccation than the wild-type strain.\t5-fold [1]\n",
      "\n",
      "['similar between human limk1', 'limk pdz domain', 'human limk2', 'identical', '81', '47']\n",
      "['47']\n",
      "The LIMK PDZ domain is 47% identical and 81% similar between human LIMK1 and human LIMK2.\t47 [1]\n",
      "\n",
      "['full_length vibrio cholerae rfah', 'similar structural properties', 'sequence identity specifically', 'longer helix α3', 'vcrfah shares 35_8', 'sequence identity', 'coli rfah', 'proteins exhibit', 'notable differences', 'not present', 'kow domain', 'evolutionarily remote', 'disulfide bridge', 'conserved mechanism', '43_6 %.', 'ecrfah ),', 'helix', 'vcrfah', 'vcrfah', 'ecrfah', 'ecrfah', 'such', 'stabilizes', 'however', 'e', 'divergence', 'despite', 'action']\n",
      "['43_6']\n",
      "The full-length Vibrio cholerae RfaH (VcRfaH) is evolutionarily remote from E. coli RfaH (EcRfaH), with a sequence identity of 43.6%. Despite this divergence, both proteins exhibit a conserved mechanism of action. VcRfaH shares 35.8% sequence identity specifically in the KOW domain with EcRfaH, and they have similar structural properties. However, there are notable differences, such as VcRfaH having a longer helix α3* and a disulfide bridge that stabilizes this helix, which are not present in EcRfaH.\t43.6 [0.93029777]\n",
      "\n",
      "['would need significantly less bleomycin', 'inhibit cell growth compared', '0_5 kpa surface', '0_5 kpa surface', 'would need', 'soft matrix', 'same effect', 'roughly 1', 'complete inhibition', 'concentration required', 'concentration needed', 'culturing fibroblasts', 'concentration', 'culturing', 'using', 'therefore', 'specifically', 'proliferation', 'plastic', 'plastic', 'plastic', 'observed', 'know', 'calculate', 'achieve', '10th', '10']\n",
      "['need 10x less']\n",
      "When culturing fibroblasts on a 0.5 KPa surface, you would need significantly less bleomycin to inhibit cell growth compared to culturing on plastic. Specifically, complete inhibition of proliferation on the soft matrix was observed at roughly 1/10th the concentration required to achieve the same effect on plastic. Therefore, if you know the concentration needed on plastic, you can calculate that you would need only 10% of that concentration when using the 0.5 KPa surface.\tYou need 10x less [0.84151248]\n",
      "\n",
      "['constrain wyckoff sites using symmetric molecular dynamics', 'lagrange multiplier \\\\( λ \\\\)', 'c λ σ λ', 'wyckoff sites within', 'constrained molecular dynamics', 'tq q q 0', 'lagrange multiplier constraints', 'equation 9 ).', 'equation 8 ),', '\\\\( σ \\\\)', '\\\\( j p', '\\\\( q', '\\\\( j', '\\\\( σ', '\\\\) represents', '0 \\\\).', '− \\\\)', 'δt )]', 'terms analytically', 'proper handling', 'involves computing', 'equations provided', 'constraints allows', 'holonomic constraint', 'holonomic constraint', 'constraint dimension', 'q', '\\\\)', '\\\\(', 'σ', 'σ', 'p', 'constraints', 'constraint', '−', '\\\\).', 'value', 'thus', 'step', 'specifically', 'respect', 'jacobian', 'indicated', 'implemented', 'implementation', 'given', 'framework', 'force', 'expressed', 'ensuring', 'elements', 'determined']\n",
      "['lagrange multiplier constraints']\n",
      "To constrain Wyckoff sites using symmetric molecular dynamics as a holonomic constraint, the constraints are implemented as Lagrange multiplier constraints. This involves computing the terms analytically at each step, as indicated by the equations provided. Specifically, the holonomic constraint can be expressed as \\( t P t tq q q 0( ) ( ) ( )σ[ ] = − = \\) (equation 8), and the force from the constraint is given by \\( J P IF ( )c λ σ λ= [ ] = − \\) (equation 9). Here, \\( J[σ] \\) represents the Jacobian of \\( σ \\) with respect to the constraint dimension and the elements of \\( q(t) \\). The value of the Lagrange multiplier \\( λ \\) can be determined by ensuring that \\( σ[q(t + Δt)] = 0 \\). Thus, the implementation of these constraints allows for the proper handling of Wyckoff sites within the framework of constrained molecular dynamics.\tLagrange multiplier constraints [1]\n",
      "\n",
      "['microglia_specific ifnar1 deletion prevents post_synaptic loss', 'pre_synaptic markers such', 'microglial ifn_i signaling', 'altering microglial activation', 'post_synaptic terminals', 'puncta density', 'not restored', 'excitatory synapses', '5xfad mice', 'synaptophysin', 'selective', 'restoration', 'psd95', 'leads', 'indicating', 'however', 'effect']\n",
      "['loss', 'attenuation']\n",
      "In 5XFAD mice, microglia-specific Ifnar1 deletion prevents post-synaptic loss by altering microglial activation, which leads to a restoration of PSD95+ puncta density. However, pre-synaptic markers such as synaptophysin+ were not restored, indicating that the effect of microglial IFN-I signaling is selective for post-synaptic terminals of excitatory synapses.\tattenuation of loss [0.85662127 0.81891737]\n",
      "\n",
      "['role promoting degradation', 'protein serrate', 'shown', 'pbe1', 'not', 'interact', 'cwc15', 'arabidopsis']\n",
      "['pbe1']\n",
      "In Arabidopsis, CWC15 has not been shown to interact with PBE1 in its role promoting degradation of the protein Serrate.\tPBE1 [1]\n",
      "\n",
      "['favor degradation according', 'degron index', 'serine', 'sequence', 'presence', 'predicted', 'position', '8', '3']\n",
      "['3']\n",
      "The presence of Serine at position -3 or -8 in the sequence is predicted to favor degradation according to the degron index.\t-3 [1]\n",
      "\n",
      "['most often displays', 'first 2_cell blastomere', '2_cell stage embryo', 'one blastomere', 'early embryo', 'small number', 'primarily originates', 'human development', 'contribute more', 'clonal imbalance', 'clonal imbalance', 'inside cells', 'founding epiblast', 'epiblast', 'cells', 'portion', 'leading', 'formation', 'epi', 'divide', 'biased', 'associated']\n",
      "['inner cell mass']\n",
      "In human development, the portion of the early embryo that most often displays a clonal imbalance is the epiblast, which primarily originates from only one blastomere of the 2-cell stage embryo. This clonal imbalance is associated with the first 2-cell blastomere to divide, which is biased to contribute more to the inside cells, leading to the formation of a small number of founding epiblast (EPI) cells.\tInner cell mass [0.8718289]\n",
      "\n",
      "['cells remain committed', 'baseline levels after', 'trigger senescence commitment', 'subsequently returned', 'human fibroblasts', 'erk signaling', 'erk activation', 'activation period', '4 days', 'senescence', 'maintained', 'even', '3']\n",
      "['4 days', '3']\n",
      "In human fibroblasts, ERK activation should be maintained for 3 to 4 days to trigger senescence commitment. Even if ERK signaling is subsequently returned to baseline levels after this activation period, the cells remain committed to senescence.\t3-4 days [1 1]\n",
      "\n",
      "['mitochondrial dna polymerase gamma', 'homozygous knock_in miras allele', 'rotarod performance', 'age compared', '12 months', 'control mice', 'mice', 'observed', 'decrease', '30']\n",
      "['30']\n",
      "In mice with a homozygous knock-in MIRAS allele of mitochondrial DNA polymerase gamma, a 30% decrease in rotarod performance is observed at 12 months of age compared with control mice.\t30% [1]\n",
      "\n",
      "['more than 72 hours before', 'pre_commitment myeloid progenitor cells', 'no longer return', 'progenitor state', 'cells', 'withdrawn', 'estrogen']\n",
      "['72 hours']\n",
      "In pre-commitment myeloid progenitor cells, estrogen can be withdrawn for more than 72 hours before the cells can no longer return to their progenitor state.\t72 hours [0.87857231]\n",
      "\n",
      "['detecting surface glycornas', 'neu5ac ).', 'n_acetylneuraminic acid', 'arpla method', 'aptamer binds']\n",
      "['neu5ac']\n",
      "In the ARPLA method for detecting surface glycoRNAs, the aptamer binds to N-acetylneuraminic acid (Neu5Ac).\tNeu5Ac [1]\n",
      "\n",
      "['could sometimes exhibit poor contact between subunits', 'including various point groups such', 'ingraham 2022 protein design paper', 'sampled geometries maintained symmetry', 'protein complexes adhered', 'specific euclidean symmetries', 'gi · sdesolve', 'gi · x1', 'underlying ode', 'strictly imposed', 'sde sampling', 'sampling procedure', 'sampling function', 'relationship sdesolve', 'invariance condition', 'icosahedral symmetries', 'held true', 'generate samples', 'desired symmetries', 'authors proposed', 'method allowed', 'group element', 'gi', 'symmetries', 'sdesolve', 'x1', 'x1', 'method', 'group', 'σisdesolve', 'tetrahedral', 'symmetrize', 'specifically', 'required', 'octahedral', 'noted', 'imposing', 'however', 'g', 'g', 'equivariant', 'ensured', 'ensure', 'dihedral', 'denoted', 'cyclic', 'considered', 'achieved']\n",
      "['symmetric sampling', 'analytical conditioning', 'sde', 'ode']\n",
      "In the Ingraham 2022 protein design paper, the authors proposed a method to symmetrize the underlying ODE/SDE sampling to ensure that the protein complexes adhered to specific Euclidean symmetries. This was achieved by imposing an invariance condition on the sampling procedure, which required that the sampling function, denoted as SDESOLVE, be equivariant to the group of symmetries (G) being considered. Specifically, they ensured that for any group element (gi) in G, the relationship SDESOLVE(gi ·X1) = gi ·SDESOLVE(X1) = σiSDESOLVE(X1) held true. This method allowed them to generate samples that strictly imposed the desired symmetries, including various point groups such as cyclic, dihedral, tetrahedral, octahedral, and icosahedral symmetries. However, they noted that while the sampled geometries maintained symmetry, they could sometimes exhibit poor contact between subunits.\tsymmetric sampling of the SDE/ODE (analytical conditioning) [0.87579169 0.79489341 1.         0.82159649]\n",
      "\n",
      "['48 well_defined lung cell types', 'lungmap human lung cellref', 'normal human lung', 'total', 'identified']\n",
      "['48']\n",
      "In the LungMAP Human Lung CellRef, a total of 48 well-defined lung cell types are identified in a normal human lung.\t48 [0.80856059]\n",
      "\n",
      "['protein location_ de novo protein folding_ innate immune response_ regulation', 'synaptonemal complex assembly_ cellular component assembly_ microtubule polymerization_ negative regulation', 'organelle localization_ chaperone cofactor_dependent refolding_ defense response', 'viral processes_ cellular component biogenesis_ establishment', 'several cellular processes', 'nervous system development', 'regulation', 'symbiont_ maintenance', 'human hnrnpcl2', 'involved', 'including']\n",
      "['nan']\n",
      "Human HNRNPCL2 is involved in several cellular processes, including:- Regulation of synaptonemal complex assembly- Cellular component assembly- Microtubule polymerization- Negative regulation of viral processes- Cellular component biogenesis- Establishment of organelle localization- Chaperone cofactor-dependent refolding- Defense response to symbiont- Maintenance of protein location- De novo protein folding- Innate immune response- Regulation of nervous system development.\tnan [0.79028494]\n",
      "\n",
      "['inactivating either col4a5', 'homozygous leukolysin knockout', 'zebrafish embryos', 'rescued', 'defects', 'col4a6', 'angiogenesis']\n",
      "['col4a5']\n",
      "In zebrafish embryos with homozygous leukolysin knockout, defects in angiogenesis can be rescued by inactivating either col4a5 or col4a6.\tcol4a5 [1]\n",
      "\n",
      "['no relevant sources', 'no information']\n",
      "['sin3 histone deacetylation']\n",
      "No information. No relevant sources.\tSIN3 histone deacetylation [0.74723331]\n",
      "\n",
      "['nfia ), indicating early glial patterning', 'comprising 25 – 31', 'gliogenic factor nuclear factor', '8 – 10 weeks', '28 – 33', 'neuroectoderm specification results', 'enables accelerated astrogliogenesis', 'gliogenic switch via', 'gliogenic switch', '3 weeks', 'robust upregulation', 'process triggers', 'neurogenic potential', 'efficient derivation', 'cell population', 'progenitor cells', 'cells', 'cells', 'subset', 'retaining', 'pdgf_aa', 'organoids', 'majority', 'leading', 'induction', 'furthermore', 'differentiation', 'differentiation', 'astroglia', 'addition']\n",
      "['normal neurogenic potential', 'accelerated astrogliogenesis']\n",
      "The induction of a gliogenic switch via the addition of PDGF-AA during neuroectoderm specification results in the efficient derivation of astroglia, comprising 25–31% of the cell population by 8–10 weeks of differentiation. This process triggers a gliogenic switch in 28–33% of the cells in the organoids at 3 weeks of differentiation, leading to robust upregulation of the gliogenic factor nuclear factor I A (NFIA), indicating early glial patterning in a subset of progenitor cells. Furthermore, it enables accelerated astrogliogenesis while retaining the neurogenic potential for the majority of cells.\taccelerated astrogliogenesis and normal neurogenic potential [0.94913647 0.95519288]\n",
      "\n",
      "['not specifically define immature neuron populations', 'identify dcx_expressing neuroblasts after childhood', 'newly generated granule cells', 'newly generated granule cells', 'adult human dentate gyrus', 'diverse human neurons', 'shown varied expression', 'widely accepted marker', 'not unique', 'unique marker', 'not', 'therefore', 'studies', 'some', 'population', 'hippocampus', 'failed', 'doublecortin', 'dg', 'detected', 'dcx', 'dcx', 'dcx', 'additionally']\n",
      "['no']\n",
      "Doublecortin (DCX) is a widely accepted marker of newly generated granule cells in the hippocampus, but it is not unique to this population. While it is detected in diverse human neurons, it does not specifically define immature neuron populations. Additionally, studies have shown varied expression of DCX in adult human dentate gyrus (DG) and some have failed to identify DCX-expressing neuroblasts after childhood. Therefore, DCX is not a unique marker of newly generated granule cells.\tNo [0.89537293]\n",
      "\n",
      "['snog1a mutant', 'physcomitrium patens', 'chromosome 8', 'causative mutation', 'located']\n",
      "['chromosome 8']\n",
      "The causative mutation of the snog1a mutant of Physcomitrium patens is located on chromosome 8.\tChromosome 8 [1]\n",
      "\n",
      "['no information']\n",
      "['70']\n",
      "No information.\t70 [0.7734341]\n",
      "\n",
      "['reactive astrocytes following tnfα treatment', 'significant upregulation', 'molecule associated', 'cells suggests', 'nad +.', 'nad', 'mediates', 'levels', 'decline', 'decline', 'cd38', 'cd38']\n",
      "['cd38']\n",
      "The molecule associated with the decline of NAD+ levels in reactive astrocytes following TNFα treatment is CD38. The significant upregulation of CD38 in these cells suggests that it mediates the decline of NAD+.\tCD38 [1]\n",
      "\n",
      "['wild type zebrafish embryos', 'display ciliary beat defects', 'reduced beat frequency', 'normal ciliary length', 'vertebrate multiciliated cells', 'include complete paralysis', 'kif9 kd embryos', 'shorter than', 'kif9 knocked', 'disrupted waveforms', 'kif9', 'relative', 'necessary', 'indicating', 'controls', 'cilia', 'cilia', 'beating', 'additionally']\n",
      "['shorter cilia']\n",
      "Relative to wild type zebrafish embryos, those with Kif9 knocked down display ciliary beat defects that include complete paralysis of cilia, reduced beat frequency, and disrupted waveforms. Additionally, the cilia in Kif9 KD embryos are shorter than those in controls, indicating that Kif9 is necessary for normal ciliary length and beating in vertebrate multiciliated cells.\tShorter cilia [0.90644396]\n",
      "\n",
      "['sialic acid_binding ig_like lectin 15', 'sialic acid moiety', 'sialic acid moieties', 't_cell surfaces decreases', 'human t_cells', '15', '15', 'siglec', 'siglec', 'removal', 'observed', 'indicating', 'essential', 'deglycosylation', 'decrease', 'comparable', 'cells', 'binding', 'binding', 'binding']\n",
      "['decreases']\n",
      "The removal of sialic acid moieties from the T-cell surfaces decreases the binding of Sialic acid-binding Ig-like lectin 15 (Siglec-15) to human T-cells. This decrease in binding is comparable to that observed with deglycosylation, indicating that the sialic acid moiety is essential for the binding of Siglec-15 to T cells.\tdecreases [1]\n",
      "\n",
      "['venular capillary marker ackr1', 've3 vascular endothelial cells', 'prenatal cellular programs', 'mac2 macrophages via', 'chemokine cxcl8 interacting', 'diseased skin', 'reemergence', 'mediated']\n",
      "['ackr1']\n",
      "The reemergence of prenatal cellular programs is mediated by Mac2 macrophages via the chemokine CXCL8 interacting with the venular capillary marker ACKR1 on VE3 vascular endothelial cells in diseased skin.\tACKR1 [1]\n",
      "\n",
      "['unc5b_as1 upstream super enhancer region', 'segment e2', 'foxp3 binds']\n",
      "['e2']\n",
      "FOXP3 binds to segment E2 of the UNC5B-AS1 upstream super enhancer region.\tE2 [1]\n",
      "\n",
      "['upregulation most associated', 'increased gene expression', 'signature characterized', 'relapse signature', 'prostate cancer', 'pro_inflammatory chemokines', 'chemokines', 'related', 'indicated', 'chemokine_enriched_gland', 'ceg']\n",
      "['cxcl17']\n",
      "The upregulation most associated with the Relapse signature in prostate cancer is related to pro-inflammatory chemokines, as indicated by the chemokine-enriched-gland (CEG) signature characterized by increased gene expression of these chemokines.\tCXCL17 [0.7]\n",
      "\n",
      "['analysis identified 367 differential morphology features', 'morphology between apoptotic', 'most changes occurring', 'cell painting assays', 'apoptotic cells', 'primarily related', 'greatest difference', 'control cells', 'cellular feature', 'organelles', 'nuclei', 'mitochondria', 'measured', 'displays']\n",
      "['mitochondria']\n",
      "The cellular feature that displays the greatest difference in morphology between apoptotic and control cells, as measured by Cell Painting assays, is primarily related to the mitochondria and nuclei. The analysis identified 367 differential morphology features, with most changes occurring in these organelles for apoptotic cells.\tmitochondria [1]\n",
      "\n",
      "['wild_type β_tubulin binds tbz more favorably', 'f200y mutant exhibits unfavorable repulsions', 'fungal β_tubulin significantly affects', 'f200y mutation likely forms', 'anti_fungal drug thiabendazole', 'occludes tbz binding', 'f200y mutation', 'unfavorable binding', 'repulsions caused', 'tbz resistance', 'tbz ).', 'residues y200', 'primarily due', 'hydrogen bond', 'contact energy', 'contact energy', 'binding pocket', '9_9 kcal', '27_6 kcal', 'mutant', 'binding', 'mol', 'mol', 'l253', 'l250', 'interactions', 'f240', 'e198', 'contributing', 'constricts', 'additionally']\n",
      "['disrupts binding']\n",
      "The F200Y mutation in fungal β-tubulin significantly affects the binding of the anti-fungal drug thiabendazole (TBZ). The wild-type β-tubulin binds TBZ more favorably with a contact energy of -9.9 kcal/mol, while the F200Y mutant exhibits unfavorable repulsions with a contact energy of +27.6 kcal/mol. This unfavorable binding is primarily due to repulsions caused by interactions with residues Y200, F240, L250, and L253 in the mutant. Additionally, the F200Y mutation likely forms a hydrogen bond with E198, which constricts the binding pocket and occludes TBZ binding, contributing to TBZ resistance.\tDisrupts binding [0.87509999]\n",
      "\n",
      "['p values indicating no significant differences among', 'bone marrow stromal cell_conditioned media', 'not significantly alter', 'unconditioned medium', 'cultured ot', 'cd8a receptor', 'adipocyte_conditioned medium', 'acm ),', '24_hour treatment', 'expression levels', 'expression', 'ucm', 'scm', 'scm', 'groups', 'comparable', 'cells', 'cd8α', 'after', '1']\n",
      "['no effect']\n",
      "Bone marrow stromal cell-conditioned media (SCM) does not significantly alter the expression of the CD8a receptor in cultured OT-1 T cells. After a 24-hour treatment with SCM, the expression levels of CD8α were comparable to those in unconditioned medium (UCM) and adipocyte-conditioned medium (ACM), with P values indicating no significant differences among the groups.\tNo effect [0.82789946]\n",
      "\n",
      "['spindle e protein', 'silkworm cells leads', 'mature piwirnas pir1712', 'longer rna signals', 'atpase_deficient e251q mutant', 'concomitant accumulation', 'reduction', 'pre_pirnas', 'pir2986', 'levels', 'lengths', 'expression', 'correspond', 'additionally']\n",
      "['decrease']\n",
      "The expression of the ATPase-deficient E251Q mutant of the Spindle E protein in silkworm cells leads to a reduction in the levels of the mature piwiRNAs piR1712 and piR2986. Additionally, there is a concomitant accumulation of longer RNA signals that correspond to the lengths of their pre-piRNAs.\tDecrease [0.86670719]\n",
      "\n",
      "['indicating significant genetic differences', 'greatest genetic differentiation parameter', 'geographical_based iranian cannabis populations', 'second fst analysis', 'highest fst value', 'high fst value', 'first fst analysis', 'southeast populations', 'southeast pair', 'fst', 'northeast', 'hemp', 'exhibited', 'east', 'compared', 'additionally', '0_17', '0_09']\n",
      "['southeast', 'northeast', 'east']\n",
      "The geographical-based Iranian cannabis populations with the greatest genetic differentiation parameter (FST) are the east and southeast populations, which exhibited the highest FST value of 0.17 in the second FST analysis when compared to hemp. Additionally, in the first FST analysis, the northeast and southeast pair had a high FST value of 0.09, indicating significant genetic differences.\tEast and southeast: northeast [0.92182685 1.         1.        ]\n",
      "\n",
      "['body fat mass loss between obese mice treated', 'absolute percent difference', '23 %.', 'glp', 'glp', '801', '1_mk', '1']\n",
      "['23']\n",
      "The absolute percent difference in body fat mass loss between obese mice treated with GLP-1-MK-801 and GLP-1 is 23%.\t23% [0.84293896]\n",
      "\n",
      "['coelicolor umbrella complex structure', 'stalk portion', 'approximate length', '300 å']\n",
      "['300 å']\n",
      "The approximate length of the stalk portion of the S. coelicolor Umbrella complex structure is about 300 Å.\t300 Å [1]\n",
      "\n",
      "['nearest topologically associated domain', 'mouse embryonic stem cells', 'most active promoters', 'lower than 0_2', 'ectopic sox2 promoter', 'mescs ),', 'contact probabilities', 'tad', 'comparable', 'boundary']\n",
      "['between 0', '0_3']\n",
      "In mouse embryonic stem cells (mESCs), the contact probabilities of most active promoters with the nearest topologically associated domain (TAD) boundary are comparable to those of the ectopic Sox2 promoter, which is lower than 0.2.\tbetween 0 and 0.3 [0.86986673 0.86034105]\n",
      "\n",
      "['poorly expressed domains inhibited', 'poorly expressed domain', 'decreased protein abundance', 'krab domain', 'some silencing', 'ml dox', 'ml allowed', 'krab fusions', 'full silencing', 'dhx16 ob_ntp', 'baz1a ddt', '1000 ng', '100 ng', 'krab function', 'krab function', 'dox concentration', 'function', 'concentration', 'such', 'related', 'observed', 'loss', 'krab_dhx16', 'krab_baz1a', 'indicating', 'increasing', 'however', 'concatenated', 'ablate']\n",
      "['eliminates function']\n",
      "When the KRAB domain is concatenated with a poorly expressed domain, such as DHX16 OB_NTP or BAZ1A DDT, it can ablate the KRAB function. This was observed at a concentration of 100 ng/mL dox, where the poorly expressed domains inhibited the KRAB function. However, increasing the dox concentration to 1000 ng/mL allowed for some silencing with KRAB-DHX16 and full silencing with KRAB-BAZ1A, indicating that the loss of function in these KRAB fusions is related to decreased protein abundance.\teliminates function [0.83393589]\n",
      "\n",
      "['cortical excitation – inhibition balance', 'interneurons experience less glutamatergic drive', 'without bmp – smad1 signaling', 'significantly reduced firing rate', 'smad1_deficient pv interneurons', 'adult mice leads', 'parvalbumin_expressing interneurons', 'reduced excitability', 'firing frequency', 'severe disruption', 'input resistance', 'current injections', 'barrel cortex', 'smad1', 'resulting', 'response', 'reduction', 'deletion', 'decrease', 'consequently', 'attributed']\n",
      "['greatly reduced']\n",
      "The deletion of SMAD1 in parvalbumin-expressing interneurons in the barrel cortex of adult mice leads to a significantly reduced firing rate in response to current injections. This reduction in firing frequency is attributed to a decrease in input resistance in the SMAD1-deficient PV interneurons. Consequently, without BMP–SMAD1 signaling, these interneurons experience less glutamatergic drive and reduced excitability, resulting in a severe disruption of the cortical excitation–inhibition balance.\tGreatly reduced [0.88219026]\n",
      "\n",
      "['no relevant sources', 'no information']\n",
      "['nan']\n",
      "No information. No relevant sources.\tnan [0.79799104]\n",
      "\n",
      "['measured dissociation constant', 'wnt5b_ror2 complex', 'between 311', '476 nm', 'zebrafish', 'kd', 'cytonemes']\n",
      "['between 311', '476 nm']\n",
      "The measured dissociation constant (Kd) for the Wnt5b-Ror2 complex in cytonemes of zebrafish is between 311 and 476 nM.\tBetween 311-476 nM [1 1]\n",
      "\n",
      "['thus ensuring low_level expression', 'truncation mutant binds', 'truncated c_terminus variant', 'novel polyadenylation event', 'new stop codon', 'maintaining cellular homeostasis', 'homeostatic basal level', 'gait target proteins', 'gait gene expression', 'gait target mrnas', 'target mrnas', 'gait complex', 'gait complex', 'vessel maintenance', 'tumor growth', 'myeloid cells', 'inhibitory effects', 'humans involves', 'high affinity', 'crucial role', 'form plays', 'expression', 'form', 'vegf_a', 'such', 'shielding', 'resulting', 'preserving', 'n1', 'n1', 'n1', 'mechanism', 'introduces', 'inhibiting', 'important', 'eprs1', 'eprs1', 'eprs1', 'eprs1', 'binding', 'achieves']\n",
      "['polyadenylation']\n",
      "The mechanism for the expression of the EPRS1(N1) form of EPRS1 in humans involves a novel polyadenylation event that introduces a new stop codon, resulting in a truncated C-terminus variant. This EPRS1(N1) form plays a crucial role in maintaining cellular homeostasis by preserving a homeostatic basal level of GAIT gene expression in myeloid cells. It achieves this by inhibiting the binding of the GAIT complex to target mRNAs, such as VEGF-A, which is important for both tumor growth and vessel maintenance. The EPRS1(N1) truncation mutant binds with high affinity to GAIT target mRNAs, shielding them from the inhibitory effects of the GAIT complex, thus ensuring low-level expression of GAIT target proteins.\tpolyadenylation [1]\n",
      "\n",
      "['optimal gdp_binding temperature', 'asgard class wukongarchaeia', '63_2 ° c', '1a protein', 'ef']\n",
      "['63_2']\n",
      "The optimal GDP-binding temperature for the EF-1A protein from Asgard class Wukongarchaeia is 63.2 °C.\t63.2 [0.9090406]\n",
      "\n",
      "['editing 50_gc substrates', 'ddd1 deaminase shows', 'preference']\n",
      "['_gc', '5']\n",
      "The Ddd1 deaminase shows a preference for editing 50-GC substrates.\t5'-GC [1.         0.77968872]\n",
      "\n",
      "['repress transcription', 'proteins screened', 'meaning', 'classified', 'bifunctional', 'activate', '12']\n",
      "['12']\n",
      "12% of the proteins screened are classified as bifunctional, meaning they can both activate and repress transcription.\t12 [1]\n",
      "\n",
      "['top 10 loci account', 'large serine recombinase cp36', 'uniquely mapped reads', 'integration site assay', 'k562 cells', '11_4']\n",
      "['11_40']\n",
      "In the integration site assay for the large serine recombinase Cp36 in K562 cells, the top 10 loci account for 11.4% of uniquely mapped reads.\t11.40% [0.91093933]\n",
      "\n",
      "['colorectal cancer_associated fibroblasts typically survive', 'approximately 50 – 80', 'platinum_based chemotherapy oxaliplatin', '2 weeks', 'cultured']\n",
      "['80', '50']\n",
      "Approximately 50–80% of colorectal cancer-associated fibroblasts typically survive at 2 weeks when cultured with the platinum-based chemotherapy oxaliplatin.\t50-80% [0.86097267 0.85177514]\n",
      "\n",
      "['non_genic genomic region windows displayed some transcription', 'coli long term evolution experiment lines', 'relaxed threshold', '50000 generations', 'e', '64_0']\n",
      "['64']\n",
      "In the E. Coli Long Term Evolution Experiment lines at 50,000 generations, 64.0% of the non-genic genomic region windows displayed some transcription at a relaxed threshold.\t64% [0.87685016]\n",
      "\n",
      "['membrane protein substrate', '5 chaperone complex', 'substrate binds', '5 tetramer', 'wild_type get3', 'substrate_bound get3', 'large excess', 'get3 component', 'vamp2', 'leading', 'get4', 'dissociation', 'added', '4']\n",
      "['get3']\n",
      "When a large excess of membrane protein substrate (Vamp2) is added to the wild-type Get3/4/5 chaperone complex, the substrate binds to the Get3 component, leading to the dissociation of the Get4/5 tetramer from the substrate-bound Get3.\tGet3 [1]\n",
      "\n",
      "['key residue interactions include polar contacts established', 'other interactions involve residues y9', 'neighboring mc1r8 residue forms', 'w12 establishes π_stacking interactions', 'interactions collectively span', 'psmd2e336 side chain', 'macrocycle mc1 binds', 'complex interaction profile', 'charge interaction', 'target psmd2', 'hydrogen bonds', 'hydrogen bond', 'binding pocket', 'backbone carbonyls', 'backbone carbonyl', 'w10', 'respectively', 'psmd2p871', 'psmd2n737', 'psmd2l873', 'psmd2k773', 'psmd2h868', 'psmd2h770', 'psmd2e878', 'mc1h7', 'indicating', 'additionally']\n",
      "['r8 residue', 'e878 residue', 'psmd2', 'mc1']\n",
      "When the macrocycle MC1 binds to its target PSMD2, key residue interactions include polar contacts established by MC1H7 with the PSMD2E336 side chain and the backbone carbonyls of PSMD2P871 and PSMD2L873. Additionally, the neighboring MC1R8 residue forms a charge interaction with PSMD2E878 and hydrogen bonds with PSMD2H868. Other interactions involve residues Y9 and W10, which hydrogen bond with the backbone carbonyl of PSMD2K773 and PSMD2N737, respectively, while W12 establishes π-stacking interactions with PSMD2H770. These interactions collectively span the binding pocket, indicating a complex interaction profile.\tR8 residue of MC1 and E878 residue of PSMD2 [0.85787663 0.82524595 1.         1.        ]\n",
      "\n",
      "['generally successful across various models', 'significant gain indicates', 'mutational landscape data', 'more pronounced benefit', 'larger gain specifically', 'adeno_associated virus 2', 'specific mention', 'provided information', 'including esm2', 'gained significantly', 'fine_tuning gives', 'fine_tuning compared', 'ankh model', 'fine_tuning', 'fine_tuning', 'ankh', 'prott5', 'predicting', 'others', 'mutations', 'fitness', 'effect', 'based']\n",
      "['similar', 'models', 'gain']\n",
      "Based on the provided information, fine-tuning gives a larger gain specifically for the Ankh model when predicting the effect of mutations in fitness of adeno-associated virus 2, as it gained significantly from fine-tuning for the mutational landscape data. While fine-tuning was generally successful across various models, including ESM2 and ProtT5, the specific mention of Ankh's significant gain indicates it had a more pronounced benefit from fine-tuning compared to the others.\tThe gain is similar for all models [0.86387075 1.         0.85938658]\n",
      "\n",
      "['rv2780 localizes mainly', 'also partially found', 'mycobacterium tuberculosis', 'minimal presence', 'mice infected', 'macrophages derived', 'endoplasmic reticulum', 'mitochondria', 'lysosome', 'er', 'cytoplasm']\n",
      "['cytoplasm']\n",
      "Rv2780 localizes mainly in the cytoplasm of macrophages derived from mice infected with Mycobacterium tuberculosis. It is also partially found in mitochondria, with minimal presence in the endoplasmic reticulum (ER) or lysosome.\tCytoplasm [1]\n",
      "\n",
      "['no discernible effect', 'n_terminal horma domain', 'yeast rev7', 'not important', 'mrx subunits', 'mrx complex', 'amino acids', 'domain', 'loss', 'interactions', 'interaction']\n",
      "['149', '1']\n",
      "The amino acids of yeast Rev7 that are not important for its interaction with the MRX complex are those in the N-terminal HORMA domain, as the loss of this domain had no discernible effect on the interactions with the MRX subunits.\t1-149 [0.81660791 0.81585126]\n",
      "\n",
      "['amp_cpg formulated ebv vaccine', 'not produced', 'not mentioned', 'igg3 ).', 'hla_expressing mice', 'gp350_specific antibodies', 'generated isotypes', 'antibody isotype', 'list', 'igm', 'igg2b', 'igg2a', 'igg1', 'ige', 'iga']\n",
      "['ige']\n",
      "The antibody isotype that is NOT produced in HLA-expressing mice by the AMP-CpG formulated EBV vaccine for gp350-specific antibodies is IgE, as it is not mentioned in the list of generated isotypes (IgA, IgM, IgG1, IgG2a, IgG2b, and IgG3).\tIgE [1]\n",
      "\n",
      "['breast cancer models via', 'bacterial cell wall component', 'prostaglandin e2_ep2 pathway', 'facilitate lung metastasis', 'shown', 'lipopolysaccharide']\n",
      "['lipopolysaccharide']\n",
      "The bacterial cell wall component that has been shown to facilitate lung metastasis in breast cancer models via the prostaglandin E2-EP2 pathway is lipopolysaccharide.\tlipopolysaccharide [1]\n",
      "\n",
      "['vault aav delivery system', 'bioconjugation system used', 'vp2 protein', 'spytag_spycatcher technology', 'int tag', 'genetically inserted', 'fusion protein', 'spytag003', 'spycatcher003_int', 'spt3', 'specifically', 'spc3_int', 'facilitate', 'designed', 'conjugation', 'attach', 'aavs', 'aav9']\n",
      "['spytag003']\n",
      "The bioconjugation system used to attach the INT tag to AAVs in the Vault AAV delivery system is the SpyTag-SpyCatcher technology. Specifically, SpyTag003 (SpT3) is genetically inserted into the VP2 protein of AAV9, and a SpyCatcher003-INT (SpC3-INT) fusion protein is designed to facilitate the conjugation.\tSpytag003 [1]\n",
      "\n",
      "['most common category', 'borgs encodes glycosyltransferases', 'rna manipulation', 'proteins involved', 'genomes', 'gene', 'dna']\n",
      "['glycosyltransferases']\n",
      "The most common category of gene in the genomes of Borgs encodes glycosyltransferases, which are proteins involved in DNA and RNA manipulation.\tGlycosyltransferases [1]\n",
      "\n",
      "['differentiation paradigm supported enhanced astrocyte maturation', 'increased branch complexity over time', 'astrocytes exhibited advanced maturation', 'most significant maturation', 'postnatal cells over', 'nuclei displayed characteristics', 'increasing organoid age', 'increasing organoid age', 'postnatal cells', 'nuclei aligning', 'previous reports', 'improvement compared', 'high percentage', 'cell type', '8_month transplants', '301 days', 'astrocytes', 'age', 'transplantation', 'transplantation', 'shows', 'notably', 'furthermore', 'additionally', '94_5', '75_8']\n",
      "['upper_layer excitatory neurons']\n",
      "The cell type that shows the most significant maturation with transplantation and increasing organoid age is astrocytes. Notably, with transplantation and increasing organoid age, a high percentage of nuclei displayed characteristics of postnatal cells, with 94.5% for the 8-month transplants. Additionally, the differentiation paradigm supported enhanced astrocyte maturation, with 75.8% of the nuclei aligning to postnatal cells over the age of 301 days, which is an improvement compared to previous reports. Furthermore, astrocytes exhibited advanced maturation and increased branch complexity over time.\tupper-layer excitatory neurons [0.8102575]\n",
      "\n",
      "['highest editing activity across three endogenous target sites', 'highest homology directed repair', 'sdd7 base editor', 'hek293t cells', 'hdr', 'efficiency', 'demonstrated']\n",
      "['cxe', 'cas9_xten_hexo1']\n",
      "The Sdd7 base editor has the highest homology directed repair (HDR) efficiency, as it demonstrated the highest editing activity across three endogenous target sites in HEK293T cells.\tCXE (Cas9-XTEN-hExo1) [0.7 0.7]\n",
      "\n",
      "['mediating nlrp3 inflammasome oligomerization', 'fisna domain', 'critical']\n",
      "['fish_specific nacht_associated', 'fisna']\n",
      "The FISNA domain is critical in mediating NLRP3 inflammasome oligomerization.\tfish-specific NACHT-associated (FISNA) [0.77117849 1.        ]\n",
      "\n",
      "['expected epigenetic changes include alterations', 'male valvular interstitial cells', 'larger polystyrene nanoparticles', 'larger particles lead', 'larger ps_np gels', 'size_dependent changes', 'expect changes', 'uniquely affect', 'acetylation states', 'acetylation state', 'ps_np', 'gels', 'acetylation', 'vic', 'therefore', 'specifically', 'methylation', 'methylation', 'methylation', 'males', 'dose_dependent', 'cultured', 'cultured']\n",
      "['increased acetylation relative', 'female vics']\n",
      "When male valvular interstitial cells (VIC) are cultured on larger polystyrene nanoparticles (PS-NP) gels, we can expect changes in both methylation and acetylation states. Specifically, the larger particles lead to dose-dependent and size-dependent changes in methylation, and they uniquely affect the acetylation state in males. Therefore, the expected epigenetic changes include alterations in both methylation and acetylation when cultured on larger PS-NP gels.\tIncreased acetylation relative to female VICs [0.87707174 0.85780007]\n",
      "\n",
      "['zdhhc5 include gsdmd', 'plasma membrane', 'crucial step', 'zdhhc5', 'pyroptosis', 'pyroptosis', 'promotes', 'palmitoylated', 'mediated', 'localization', 'lead', 'gsdme_fl', 'gsdmb_fl', 'gene']\n",
      "['gsdmd']\n",
      "The gene(s) that can lead to pyroptosis when mediated by ZDHHC5 include GSDMD, GSDMB-FL, and GSDME-FL, as they can be palmitoylated by ZDHHC5, which promotes their localization to the plasma membrane, a crucial step for pyroptosis.\tGSDMD [1]\n",
      "\n",
      "['no information']\n",
      "['nan']\n",
      "No information.\tnan [0.79799104]\n",
      "\n",
      "['no information given']\n",
      "['none', 'above']\n",
      "No information given.\tNone of the above [0.82518034 0.77231031]\n",
      "\n",
      "['increase antibody neutralization potency', 'engineered antibody 2130', '1 spike protein', 'sars_cov2 ba', '1', 'tl59e', 'sl33a', 'sl32a', 'shown', 'part', 'mutations', 'gh112e', '112', '0114']\n",
      "['s446g']\n",
      "The mutations in the SARS-CoV2 BA.1 spike protein that have been shown to increase antibody neutralization potency are GH112E, SL32A, SL33A, and TL59E, as part of the engineered antibody 2130-1-0114-112.\tS446G [0.7]\n",
      "\n",
      "['pull gly55 towards gly482', 'cdr2 loop closer', 'spike domain', 'his56 pulls', 'computational models', 'sars_cov', 'rbd', 'rbd', 'observed', 'mutation', 'gly56', '2']\n",
      "['g56h']\n",
      "The mutation of Gly56 to His56 pulls the CDR2 loop closer to the RBD of the spike domain of SARS-CoV-2, as it has been observed to pull Gly55 towards Gly482 of the RBD in computational models.\tG56H [0.7]\n",
      "\n",
      "['strongest interacting mutant measured', 'placing hydrophobic side chains', 'v91m also strengthen', 'other mutations such', 'non_canonical hydrophobic pocket', 'most significant increase', 'same pocket', 'yeast pbs2', 'specifically noted', 'close contact', 'interaction strength', 'interaction', 'interaction', 'v91l', 'side_chain', 'sh3', 'positioning', 'mutation', 'increases', 'i87w', 'i87w', 'h86w', 'described']\n",
      "['p97a']\n",
      "The mutation in yeast Pbs2 that increases its interaction with SH3 is I87W, as it is described as the strongest interacting mutant measured, positioning its side-chain in close contact with the non-canonical hydrophobic pocket. Other mutations such as H86W, V91L, and V91M also strengthen the interaction by placing hydrophobic side chains in the same pocket, but I87W is specifically noted for having the most significant increase in interaction strength.\tP97A [0.7]\n",
      "\n",
      "['mutant drosophila smn d20v protein compared', 'greatest reduction', 'smn', 'wt', 'wild_type', 'smd1', 'proteins', 'g', 'f', 'e', 'd2', 'association']\n",
      "['smd1']\n",
      "The proteins that have the greatest reduction in association with the mutant drosophila SMN D20V protein compared to the wild-type (WT) SMN are SmD1, D2, E, F, and G.\tSmD1 [1]\n",
      "\n",
      "['hdac1 catalytic site', 'narrow tunnel leading', 'reaches halfway', 'nearby vestibule', 'likely arg312', 'arg314 occupies', '2c loops', 'tunnel', 'tunnel', 'residue', 'kbtbd4_pr', 'inserted', 'expansion', 'entrance', 'contributes', 'additionally', '2b']\n",
      "['arg312']\n",
      "The residue of KBTBD4-PR that contributes to the expansion of the 2b-2c loops and is inserted into the narrow tunnel leading to the HDAC1 catalytic site is likely Arg312, as it reaches halfway into the tunnel. Additionally, Arg314 occupies a nearby vestibule at the tunnel's entrance.\tArg312 [1]\n",
      "\n",
      "['promote large ribosomal subunit assembly even', 'e198a )), stimulating ribosome biogenesis', 'ribosomal rna modification enzyme rlmb', 'catalytically dead mutant', 'ribosome assembly', 'native rlmb', 'same extent', 'modification_independent function', 'rlmb', 'rlmb', 'supporting', 'role', 'nearly', 'found', 'expressed', 'additionally']\n",
      "['rlmb']\n",
      "The ribosomal RNA modification enzyme RlmB has been found to promote large ribosomal subunit assembly even when expressed as a catalytically dead mutant (RlmB(E198A)), stimulating ribosome biogenesis to nearly the same extent as the native RlmB. Additionally, RlmB has a modification-independent function, further supporting its role in ribosome assembly.\tRlmB [1]\n",
      "\n",
      "['flies lacking rev3 require', 'significantly more sensitive', 'rev3 ∆ mutants', 'dna alkylation induced', 'deoxycytidyl transferase activity', 'rev1 ∆ mutants', 'rev3 domain', 'drosophila lacking', 'polymerase zeta', 'methyl methanesulfonate', 'greatest increase', 'rev1 protein', 'tolerate mms', 'mms compared', 'activity', 'rev1', 'rev1', 'mms', 'supported', 'subunit', 'sensitivity', 'leads', 'knocked', 'information']\n",
      "['dtd']\n",
      "The subunit of the Rev1 protein that, when knocked out in Drosophila lacking the Rev3 domain of polymerase zeta, leads to the greatest increase in sensitivity to DNA alkylation induced by methyl methanesulfonate (MMS) is the deoxycytidyl transferase activity of REV1. This is supported by the information that flies lacking REV3 require this activity of REV1 to tolerate MMS, and that rev1∆ mutants are significantly more sensitive to MMS compared to rev3∆ mutants.\tDTD [0.7]\n",
      "\n",
      "['2_07 ± 0_56 microns', 'bud40 material produces', 'thickest fibers', 'average thickness']\n",
      "['bud40']\n",
      "The BUD40 material produces the thickest fibers, with an average thickness of 2.07± 0.56 microns.\tBUD40 [1]\n",
      "\n",
      "['real_time functional magnetic resonance imaging neurofeedback', 'achieved symptom reduction regardless', 'major depressive disorder could', 'regulation task could', 'salience network regions', 'found between subtypes', 'executive control areas', 'whole_brain activation patterns', 'whole_brain activation patterns', 'exhibited increased activation', 'different activation patterns', 'influence treatment efficacy', 'certain brain areas', 'neurofeedback training', 'suppressive activation', 'treatment outcomes', 'brain response', 'significant associations', 'self_regulation training', 'reg_b subtype', 'madrs scores', 'inter_individual differences', 'feedback signals', 'feedback signals', 'clinical response', 'negative responses', 'negative', 'whether', 'variations', 'variability', 'specifically', 'positive', 'patients', 'offset', 'instance', 'individuals', 'indicating', 'explained', 'contributing', 'changes', 'additionally', 'absence']\n",
      "['none', 'above']\n",
      "The inter-individual differences in clinical response to real-time functional magnetic resonance imaging neurofeedback in patients with major depressive disorder could be explained by variations in whole-brain activation patterns during self-regulation training. Specifically, significant associations were found between subtypes of whole-brain activation patterns during neurofeedback training and changes in MADRS scores, indicating that different activation patterns can influence treatment efficacy. For instance, individuals in the REG-B subtype, who exhibited increased activation in executive control areas and salience network regions, achieved symptom reduction regardless of whether their brain response to feedback signals was positive or negative. Additionally, the absence of suppressive activation during the regulation task could be offset by negative responses to feedback signals in certain brain areas, further contributing to the variability in treatment outcomes.\tNone of the above [0.85883327 0.83888223]\n",
      "\n",
      "['rna binding protein unkempt', 'intrinsically disordered region', 'transcriptional activity', 'silences', 'residue', 'mutated', 'l522', 'alanine']\n",
      "['l522']\n",
      "The residue in the intrinsically disordered region of RNA binding protein Unkempt that silences its transcriptional activity when mutated to alanine is L522.\tL522 [1]\n",
      "\n",
      "['canonical eif4 – eif3 initiation mechanism', 'pathway primarily responsible', 'translation initiation', 'multitail mrnas']\n",
      "['eif4e_eif4g']\n",
      "The pathway primarily responsible for translation initiation of multitail mRNAs is the canonical eIF4–eIF3 initiation mechanism.\teIF4E-eIF4G [0.7]\n",
      "\n",
      "['construct showed significant enhancements', 'resisting degradation compared', 'rna degradation', 'three branches', 'synergistic incorporation', 'other constructs', 'luminescence signals', 'highest resistance', 'branched poly', 'construct', 'poly', 'stem', 'ps', 'oligos', 'oligo', 'indicating', 'effectiveness', 'demonstrates', '2moe']\n",
      "['modified branch poly']\n",
      "The poly(A) oligo that demonstrates the highest resistance to RNA degradation is the construct with three branches and synergistic incorporation of 2MOE and PS on both the stem and branched poly(A) oligos. This construct showed significant enhancements in luminescence signals, indicating its effectiveness in resisting degradation compared to other constructs.\tmodified branch poly(A) [0.92264165]\n",
      "\n",
      "['pan_kras non_covalent inhibition', 'impose selectivity constraints', 'three residues', 'ras isoforms', 'g domains', 'evolutionary divergence', 's122 contribute', 'orthosteric effects', 's122', 'effects', 'p121', 'p121', 'involved', 'indirect', 'h95', 'h95', 'allosteric']\n",
      "['95', '122', '121']\n",
      "The three residues with evolutionary divergence in the G domains of RAS isoforms that impose selectivity constraints on pan-KRAS non-covalent inhibition are H95, P121, and S122. H95 is involved in orthosteric effects, while P121 and S122 contribute through allosteric (or indirect) effects.\t95, 121, 122 [0.86307757 0.86467532 0.88927729]\n",
      "\n",
      "['undergone endurance training', 'rattus norvegicus rats', 'transcription factor', 'skeletal muscle', 'mef2c', 'heart', 'found', 'enriched']\n",
      "['mef2c']\n",
      "The transcription factor that has been found to be enriched in the heart and skeletal muscle of Rattus norvegicus rats that have undergone endurance training is MEF2C.\tMEF2C [1]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "answer_store=[]\n",
    "print(\"Embedding Answers\")\n",
    "for i in range(len(answer)):\n",
    "    print(i/len(answer)*100, end=\"\")\n",
    "    print(\"\\r\", end=\"\")\n",
    "    answer_store.append(embedding_answers(answer[i], ideal[i], custom_stopwords, english_words))\n",
    "    print(answer[i]+\"\\t\"+ideal[i]+\" \"+str(answer_store[i])+\"\\n\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "cmbagent_env",
   "language": "python",
   "name": "cmbagent_env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
